client/src/components/Dataset/DatasetView.vue

```diff
@@ -0,0 +1,373 @@
+                        <iframe
+                            :src="displayUrl"
+                            title="galaxy dataset display frame"
+                            class="dataset-preview-iframe"
+                            frameborder="0"
+                            @load="iframeLoading = false"></iframe>
+                    </div>
+                </BTab>
+                <BTab title="Details">
+                    <DatasetDetails :dataset-id="datasetId" />
+                </BTab>
+                <BTab title="Visualize">
+                    <VisualizationsList :dataset-id="datasetId" />
+                </BTab>
+                <BTab title="Edit">
+                    <DatasetAttributes :dataset-id="datasetId" />
+                </BTab>
+                <BTab v-if="showError" title="Error Report">
+                    <DatasetError :dataset-id="datasetId" />
+                </BTab>
+            </BTabs>
+        </div>
+    </div>
+    <!-- Loading state indicator -->
+    <div v-else class="loading-message">Loading dataset details...</div>
+</template>
+
+<style scoped>
+.dataset-view {
+    height: 100%;
+}
+
+.dataset-header {
+    margin-bottom: 1rem;
+}
+
+.item-identifier {
+    display: inline-flex;
+    align-items: center;
+    max-width: 100%;
+    flex-wrap: wrap;
+}
+
+.dataset-name {
+    margin-bottom: 0;
+}
+
+.hid-box {
+    margin-right: 0.5rem;
+    white-space: nowrap;
+}
+
+.dataset-title {
+    font-weight: bold;
+    white-space: normal;
+    word-break: break-word;
+}
+
+.state-icon {
+    display: inline-flex;
+    align-items: center;
+}
+
+.state-pill {
+    display: inline-flex;
+    align-items: center;
+    padding: 0.15rem 0.5rem;
+    border-radius: 0.25rem;
+    font-size: 0.85rem;
+    white-space: nowrap;
+}
+
+.state-running,
+.state-upload {
+    background-color: #cff4fc;
+    color: #055160;
+}
+
+.state-queued,
+.state-new {
+    background-color: #e2e3e5;
+    color: #41464b;
+}
+
+.state-ok {
+    background-color: #d1e7dd;
+    color: #0f5132;
+}
+
+.state-error,
+.state-failed_metadata {
+    background-color: #f8d7da;
+    color: #842029;
+}
+
+.header-details {
+    margin-top: 0.5rem;
+    padding-left: 1rem;
+    animation: slideDown 0.25s ease;
+}
+
+@keyframes slideDown {
+    from {
+        opacity: 0;
+        transform: translateY(-10px);
+    }
+    to {
+        opacity: 1;
+        transform: translateY(0);
+    }
+}
+
+.dataset-tabs-container {
+    min-height: 0;
+}
+
+.dataset-tabs-container :deep(.nav-tabs) {
+    flex-shrink: 0;
+}
+
+.dataset-tabs-container :deep(.tab-content) {
+    flex-grow: 1;
+    min-height: 0;
+}
+
+.dataset-tabs-container :deep(.tab-pane) {
+    height: 100%;
+    width: 100%;
+    display: flex;
+    flex-direction: column;
+}
+
+.dataset-tabs-container :deep(.iframe-card) {
+    flex-grow: 1;
+    display: flex;
+    flex-direction: column;
+    padding: 0;
+    overflow: hidden;
+}
```

Comment:
Customization is fine, but we should introduce a reusable Tabs component so this can be reused going forward.

Reply:
This is not so critical right now that hardcode colors have been replaced with theme-colors. Thx.